include/database.py

```diff
@@ -1,18 +1,30 @@
 # database.py
 #
 # This module provides a wrapper for basic database functionality.
+import sys
 import psycopg2
 
+
 def select(connectionString, sql, parameters):
+
     # Open the connection
-    connection = psycopg2.connect(connectionString)
-    cursor = connection.cursor()
+    try:
+        connection = psycopg2.connect(connectionString, connect_timeout=5)
+        cursor = connection.cursor()
+
+        # Execute the query, note the rows
+        cursor.execute(sql, parameters)
+
+        rows = cursor.fetchall()
+
+        # Clean-up and return
+        cursor.close()
+        connection.close()
+
+        return rows
 
-    # Execute the query, note the rows
-    cursor.execute(sql, parameters)
-    rows = cursor.fetchall()
+    except psycopg2.OperationalError as err:
+        sys.stderr.write(f'Error Connecting: Connection Timed out {err} ')
```

Comment:
Two things. First, since operational error can be many different things we can't assume that it's a timeout, so check with the psycopg2 documentation and see if there is a way to tell. If there isn't then go with a general error (e.g., "An error occurred connecting to the database: `{err}`"). 

Second, remember that we need to rethrow an exception after printing the meaningful. That means you still need to go through the rest of the scripts and make sure they can gracefully handle the errors (e.g., by catching it and exiting with the error flag set).

Reply:
Can I use `raise SystemExit` to exit if exception is raised?

Reply:
That would be fine for the scripts calling `select`, just make sure that the error code is set for `system.exit`.